python/cudf/cudf/utils/gpu_utils.py

```diff
@@ -53,12 +56,32 @@ def validate_setup():
     except CUDARuntimeError as e:
         if e.status in notify_caller_errors:
             raise e
+
+        # We must distinguish between "CPU only" and "the driver is
+        # insufficient for the runtime".
+        if e.status == cudaError_t.cudaErrorInsufficientDriver:
+            try:
+                status, _ = cudaDriverGetVersion()
+            except RuntimeError:
+                # cudaDriverGetVersion() can raise a RuntimeError
+                # when ``libcuda.so`` is missing.
+                # We don't want this to propagate up to the user.
+                warnings.warn(str(e))
+            finally:
+                if status != cudaError_t.cudaSuccess:
+                    # Error while getting the driver version
+                    raise CUDARuntimeError(status)
```

Comment:
Can we `raise from e` here so that we get the full tb in future debugging?

Reply:
```suggestion
                    raise CUDARuntimeError(status) from e
```